qiskit_machine_learning/algorithms/classifiers/pegasos_qsvc.py

```diff
@@ -0,0 +1,285 @@
+        self._alphas = {}
+
+        t_0 = datetime.now()
+        # training loop
+        for step in range(1, self._num_steps + 1):
+            # for every step, a random index (determining a random datum) is fixed
+            i = np.random.randint(0, len(y))
+
+            value = self._compute_weighted_kernel_sum(i, X, training=True)
+
+            if (self._label_map[y[i]] * self.C / step) * value < 1:
+                # only way for a component of alpha to become non zero
+                self._alphas[i] = self._alphas.get(i, 0) + 1
+
+        self._fit_status = True
+
+        logger.debug("fit completed after %s", str(datetime.now() - t_0)[:-7])
+
+        return self
+
+    # pylint: disable=invalid-name
+    def predict(self, X: np.ndarray) -> np.ndarray:
+        """
+        Perform classification on samples in X.
+
+        Args:
+            X: Features. For a callable kernel shape (m_samples, n_features), for a precomputed
+               kernel shape (m_samples, n_samples), where m denotes the set to be predicted and n the
+               size of the training set. In that case, the kernel values in X have to be calculated
+               with respect to the elements of the set to be predicted and the training set.
+
+        Returns:
+            y_pred: Shape (n_samples), the predicted class labels for samples in X.
+
+        Raises:
+            QiskitMachineLearningError:
+                - predict is called before the model has been fit.
+            ValueError:
+                - Pre-computed kernel matrix has the wrong shape and/or dimension.
+        """
+        if not self._fit_status:
+            raise QiskitMachineLearningError("The PegasosQSVC has to be fit first")
+        if np.ndim(X) != 2:
+            raise ValueError("X has to be a 2D array")
+        if self._precomputed and self._n_samples != X.shape[1]:
+            raise ValueError(
+                "For a precomputed kernel, X should be in shape (m_samples, n_samples)"
+            )
+
+        t_0 = datetime.now()
+        y = np.zeros(X.shape[0])
+        for i in range(X.shape[0]):
+            value = self._compute_weighted_kernel_sum(i, X, training=False)
+
+            if value > 0:
+                y[i] = self._label_pos
+            else:
+                y[i] = self._label_neg
+
+        logger.debug("prediction completed after %s", str(datetime.now() - t_0)[:-7])
+
+        return y
+
+    def _compute_weighted_kernel_sum(self, index, X, training) -> float:
+        """Helper function to compute the weighted sum over support vectors used for both training
+        and prediction with the Pegasos algorithm.
+
+        Args:
+            index: fixed index distinguishing some datum
+            X: Features
+            training: flag indicating whether the loop is used within training or prediction
+
+        Returns:
+            float: weighted sum of kernel evaluations employed in the Pegasos algorithm
```

Comment:
The return type, in the docs, gets picked out from the method signature when there are typehints. So no need to replicate here otherwise it makes maintenance harder when its duplicated. On the subject of typehints the parameters in the above signature are  missing types.

Reply:
Type hints are added, return type from the docstring is removed.